### PR TITLE
Feature/gazebo gps

### PR DESCRIPTION
**Pull request by LorenzMeier**
@nicolaerosia For diff against upstream.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,12 +3,13 @@ cmake_minimum_required(VERSION 2.8.8 FATAL_ERROR)
 find_package( PkgConfig REQUIRED)
 find_package(OpenCV REQUIRED)
 find_package(gazebo REQUIRED)
-pkg_check_modules( EIGEN3 REQUIRED eigen3)
```
Comment (LorenzMeier) [inline CMakeLists.txt]: This won't work.

File: include/liftdrag_plugin/liftdrag_plugin.h [hunk 2]
```diff
@@ -1,140 +0,0 @@
-/*
```
Comment (LorenzMeier) [inline include/liftdrag_plugin/liftdrag_plugin.h]: The plugin is not actually installed on Mac OS. We still need to build it.

**Comment by LorenzMeier** [comment]
Hmm - given that you changed your earlier PR and we addressed a couple of issues there, would you be willing to rebase your other changes on that version?

When you do, please also leave the lift_drag plugin in (for 6 and 7) and please note my comments about the issues with the CMake build on OS X.

File: include/liftdrag_plugin/liftdrag_plugin.h (hunk 2 above)
Comment (nicolaerosia) [inline include/liftdrag_plugin/liftdrag_plugin.h]: Please see https://github.com/PX4/sitl_gazebo/pull/13#issuecomment-206120124

**Comment by nicolaerosia** [comment]
I can rebase, sure.
Can you please check if the plugin is there?
Try searching for '_LiftDrag_', I have it in ./lib64/gazebo-7/plugins/libLiftDragPlugin.so

**Comment by LorenzMeier** [comment]
@nicolaerosia Indeed, upon looking again I found it in `/usr/local/lib/gazebo-6.5/plugins/` on Mac OS. So we need the CMake logic to link against it on Mac.

**Comment by LorenzMeier** [comment]
@nicolaerosia Now we need to rework this on top of master and should be able to bring most things in.

**Comment by nicolaerosia** [comment]
Do you agree with those Hil\* removal?
I've removed them because I didn't find how to use them in this fork.
Seems to be used in Rotors ?

**Comment by LorenzMeier** [comment]
@nicolaerosia Yes, can be stripped.

**Comment by nicolaerosia** [comment]
I have modified the sdf which is the wrong way to go since it is generated.
I will modify the base to add gps.
We should remove the sdf from git and add a build step to generate sdf when building.

**Comment by LorenzMeier** [comment]
@nicolaerosia Ping me when this is ready for another go. I'm set up to verify things now very quickly.

**Comment by nicolaerosia** [comment]
@LorenzMeier please take a look.
I've rebased everything and rewritten history to be cleaner.

**Comment by LorenzMeier** [comment]
Failed on Mac OS:

```
Error [parser.cc:286] Error finding file [/Users/lorenzmeier/src/Firmware/Tools/sitl_gazebo/models/iris/iris.sdf].
Error [parser.cc:689] Unable to read file[/Users/lorenzmeier/src/Firmware/Tools/sitl_gazebo/models/iris/iris.sdf]
Error [parser.cc:769] Error reading element <world>
Error [parser.cc:394] Unable to read element <sdf>
Error [parser_urdf.cc:3406] Unable to call parseURDF on robot model
Error [parser.cc:309] parse as old deprecated model file failed.
[Err] [Server.cc:336] Unable to read sdf file[../worlds/iris.world]
```

**Comment by LorenzMeier** [comment]
The SDF generation must now be missing / broken.

**Comment by nicolaerosia** [comment]
Make sdf?

**Comment by LorenzMeier** [comment]
Or maybe it was broken all along and only now became apparent. Let me check..

**Comment by LorenzMeier** [comment]
Yep, that did it. Let me fix our automation scripts, thanks!

**Comment by LorenzMeier** [comment]
Position hold is pretty bad. So something is not yet quite right with GPS.

**Comment by nicolaerosia** [comment]
Ok.  I will compare the implementations.

**Comment by LorenzMeier** [comment]
I spent some time on this, also trying to ramp up the accuracy of the GPS sim data, but wasn't able to pinpoint or fix it yet.

**Comment by LorenzMeier** [comment]
It looks like the GPS data is scaled incorrectly. The error in meters is much less in meters than visible in Gazebo.

**Comment by LorenzMeier** [comment]
It looks like the version on master has similar issues. It might not be related to this PR after all.

**Comment by LorenzMeier** [comment]
@nicolaerosia I fixed the issues in the core simulation, the weirdnesses with this still remain:
- Magnetic declination is off
- GPS glitches lead the system to believe its in a position its not really. Unless "real GPS" glitches they lead to a constant offset.

**Comment by nicolaerosia** [comment]
@LorenzMeier what issues are left? I can also take a look, but I don't have experience with PX4
"Magnetic declination is off" off as in off by an offset? Or disabled?
"GPS glitches" caused by Gazebo? Maybe there is a problem related to reference frames?

**Comment by LorenzMeier** [comment]
@nicolaerosia Sorry for the silence in the last months - we were extremely busy keeping the basic SITL stable and operational. We're through that though and now want to bring these changes in. After fixing a lot of other things I'm finally in the shape to validate this PR properly.

**Comment by TSC21** [comment]
@LorenzMeier should we bring this to a testable condition?

**Comment by LorenzMeier** [comment]
We did validation but still found differences. I think doing a fresh stab using this as guidance could make sense.

**Comment by TSC21** [comment]
Agree!

**Comment by LorenzMeier** [comment]
Stale, giving up.

**Comment by TSC21** [comment]
@LorenzMeier actually we already have a GPS plugin on master and working https://github.com/PX4/sitl_gazebo/pull/142.

**Comment by LorenzMeier** [comment]
Yes, but that's duplicating the standard GPS plugin of Gazebo 8. We should try to avoid duplicating what is already available (in likely more sophistication) as part of Gazebo.

**Comment by LorenzMeier** [comment]
I'm also just revising it to have less noise. The simulated noise level is pretty unrealistic.

**Comment by TSC21** [comment]
> Yes, but that's duplicating the standard GPS plugin of Gazebo 8. We should try to avoid duplicating what is already available (in likely more sophistication) as part of Gazebo.

Ok I didn't notice that. For the case of Gazebo 7, would it make sense then to us this one while using the std Gazebo GPS plugin of v8 , with the cmake detecting the Gazebo version?

> I'm also just revising it to have less noise. The simulated noise level is pretty unrealistic.

Yeah I'm aware of it and we are seeing the consequences of it in HIL flight with EKF2 activated on PX4 side.

**Comment by TSC21** [comment]
@LorenzMeier maybe I can help you out by rebasing https://github.com/PX4/sitl_gazebo/pull/82? @jgoppert is really busy atm so maybe I can lend a hand.

**Comment by LorenzMeier** [comment]
Yes, that would be great! I've pushed a hotfix, but I would not be surprised if the solution by James is more thorough.

**Comment by TSC21** [comment]
Sure. I think we can test both approaches on this case and check which ones goes better.

**Pull request closed without merging**